### PR TITLE
feat: add function similarity_search_limit_score to vectorstores.redis

### PR DESCRIPTION
**Pull request by hx23840**
# Description
***
Add function similarity_search_limit_score and similarity_search_with_score

# How to use
***
``
rds = Redis.from_existing_index(embeddings, redis_url="redis://localhost:6379", index_name='link')

rds.similarity_search_limit_score(query, k=3, score=0.2)

rds.similarity_search_with_score(query, k=3)
``

### CODE REVIEW

File: langchain/vectorstores/redis.py [hunk 1]
```diff
@@ -84,6 +84,33 @@ def add_texts(
     def similarity_search(
         self, query: str, k: int = 4, **kwargs: Any
     ) -> List[Document]:
+        docs_and_scores = self.similarity_search_with_score(query, k=k)
+        return [doc for doc, _ in docs_and_scores]
+
+    def similarity_search_limit_score(
+        self, query: str, k: int = 4, score_thread: float = 0.2, **kwargs: Any
```
Comment (hwchase17) [inline langchain/vectorstores/redis.py:91]: `score_thread` -> `score_threshold`?

File: langchain/vectorstores/redis.py [hunk 2]
```diff
@@ -84,6 +84,33 @@ def add_texts(
...
+    ) -> List[Document]:
+        docs_and_scores = self.similarity_search_with_score(query, k=k)
+
+        docs = []
+        for doc in docs_and_scores:
```
Comment (hwchase17) [inline langchain/vectorstores/redis.py:96]: `for doc, score in docs_and_scores:` more readable

File: langchain/vectorstores/redis.py [hunk 3]
```diff
@@ -84,6 +84,33 @@ def add_texts(
     def similarity_search(
         self, query: str, k: int = 4, **kwargs: Any
     ) -> List[Document]:
+        docs_and_scores = self.similarity_search_with_score(query, k=k)
+        return [doc for doc, _ in docs_and_scores]
+
+    def similarity_search_limit_score(
+        self, query: str, k: int = 4, score_thread: float = 0.2, **kwargs: Any
+    ) -> List[Document]:
```
Comment (hwchase17) [inline langchain/vectorstores/redis.py:92]: would be good to add docstring explaining if documents are kept if the score is less than score_threshold (could also be thought of as above, without knowing more)

File: langchain/vectorstores/redis.py (hunk 1 above)
Comment (hx23840) [reply langchain/vectorstores/redis.py:91]: I will fix it

File: langchain/vectorstores/redis.py (hunk 3 above)
Comment (hx23840) [reply langchain/vectorstores/redis.py:92]: Okay, I will add the description

File: langchain/vectorstores/redis.py (hunk 2 above)
Comment (hx23840) [reply langchain/vectorstores/redis.py]: Okay, it has been modified

**Review by hwchase17 — APPROVED** [review]

**Pull request merged**